src/resourceIndex.js

```diff
@@ -100,18 +101,14 @@ async function updateResourceVersions() {
   } else {
     const [Bucket, Key] = ['nextstrain-inventories', 'resources.json.gz'];
     utils.verbose(`Updating available resources index from s3://${Bucket}/${Key}`);
-    const S3 = new AWS.S3();
     try {
-      const newETag = (await S3.headObject({Bucket, Key}).promise()).ETag;
+      const newETag = (await fetch(await signedUrl({bucket:Bucket, key:Key, method: 'HEAD'}), {method: 'HEAD'}))
+        .headers.get('etag'); // value is enclosed in double quotes, but that doesn't matter for our purposes
       if (newETag && newETag === eTag) {
         utils.verbose("Skipping available resource update as eTag hasn't changed");
         return;
       }
-      /* NOTE: in AWS SDK v2 response.Body is a buffer; in v3 it's a stream */
-      const newResources = JSON.parse(
-        await S3.getObject({Bucket, Key}).promise()
-          .then((response) => gunzip(response.Body))
-      );
+      const newResources = JSON.parse(await gunzip(await (await fetch(await signedUrl({bucket:Bucket, key:Key, method: 'GET'}), {method: 'GET'})).buffer()));
```

Comment:
This doesn't check the response status code, which can lead to unclear warnings like "Resource updating failed: incorrect header check" when the actual issue was an error response. It could even lead to the wrong data structure to be stored in `newResources` / `resources`. Best to check the response status explicitly for all requests.

Separately, since this is now using `fetch`, relying on caching and revalidation is pretty natural, as [demonstrated by this patch](https://gist.github.com/tsibley/2a090228891111162537856e5ef8ae05).

Reply:
Thanks! Response code of 200 is now enforced. I'm still using eTag checking.